### PR TITLE
fix!: disable NATS/Redis in CU by default

### PR DESCRIPTION
**Pull request by ben-foxmoore**
These dependencies are enabled during Chart Testing thanks to the values files added to the two `ci` folders, now included in both CU-CP and CU-UP charts.

### CODE REVIEW

**Comment by ben-foxmoore** [comment]
CI automatically uses the new `ci-values.yaml` file: https://github.com/accelleran/helm-charts-ng/actions/runs/8111495676/job/22170866286?pr=125#step:12:40

**Review by Wielewout — COMMENTED** [review]
We would maybe also need to look into the bootstrap hostnames?

**Comment by ben-foxmoore** [comment]
Should we go with the practical `drax-...` or the perhaps more correct empty values? 🤔

**Comment by Wielewout** [comment]
Also the same should be done maybe in cell-wrapper and E2 at some point then (will be a breaking change, but the application will break for sure with the next release).

**Comment by ben-foxmoore** [comment]
Also, this change could also be made to E2 and CW, once we're happy with the approach.

**Comment by Wielewout** [comment]
I think there are two things we should take into account/be aware of:

* Make sure that the defaults give us a working installation. If we can't guarantee this the user should at least get guidance on what's going wrong with a clear error message.
* Each test scenario that we would add at some point would need additional settings with the current approach as the default isn't giving us a working installation anymore.

Users won't be getting a working installation with the current state of the PR as empty default values would result in nats and redis service references `{{ $.Release.Name }}-nats` and `{{ $.Release.Name }}-redis-master`. We could go for `drax-nats` and `drax-redis-master` which I'm personally not a fan of as this restricts the installation to a specific release name in another chart (I would prefer this to remain decoupled completely).
So the best I can come up with is checking if nats/redis is enabled or not and when it's disabled require a value to be filled in within the bootstrap hostnames. That way we can disable it by default, but we require the integration team to fill in the hostnames (at which point I don't see why the hassle to fill in 2 more values to be honest :sweat_smile:).

If we change the default we can maybe provide a values file `values-stand-alone.yaml` to give an easy way to get a stand-alone cu install going? This would essentially be the same as the one values file we test with in the workflow.

**Comment by ben-foxmoore** [comment]
Looking at the templates/charts, I think this actually needs a change in our `common` chart, right? We would need to adapt this to check if `redis.enabled` is `true` before defaulting to the default naming:
```
{{- define "accelleran.common.bootstrap.redis.hostname" -}}
{{- $ := get . "top" | required "The top context needs to be provided to common bootstrap redis hostname" -}}
{{- if (($.Values.bootstrap).redis).enabled -}}
{{- tpl (toString ((($.Values.bootstrap).redis).hostname | default (printf "%s-redis-master" $.Release.Name)) $ -}}
{{- end -}}
{{- end -}}
```

It feels like it's starting to become quite magical, but I'm not sure I see a better solution... 🤔

**Comment by Wielewout** [comment]
Indeed, that way it's luckily only one place to do it.
The same should maybe also be done for kafka to make everything at least consistent for the bootstrap templates.

**Comment by ben-foxmoore** [comment]
Blocked by #142

**Comment by Wielewout** [comment]
Can be merged after cu-cp and cu-up are updated to use the latest common chart where the changes from #142 are included:

* [x] #249 
* [x] #250

**Review by Wielewout — APPROVED** [review]

**Pull request merged**
